packages/sdk/src/types/Loyalties.ts

```diff
@@ -497,3 +497,166 @@ export interface LoyaltyPointsTransfer {
 	code: string
 	points: number
 }
+
+// 0 Level types
+
+export type LoyaltiesGetEarningRuleResponseBody = EarningRuleBase & {
+	validation_rule_id: string | null
+	updated_at: string | null
+	active: boolean
+}
+
+export type LoyaltiesEnableEarningRulesResponseBody = EarningRuleBase & {
+	updated_at: string | null
+	active: true
+}
+
+export type LoyaltiesDisableEarningRulesResponseBody = EarningRuleBase & {
+	updated_at: string | null
+	active: false
+}
+
+// Domain types
+export interface EarningRuleBase {
+	id: string
+	created_at: string
+	loyalty: EarningRuleLoyaltyFixed | EarningRuleLoyaltyProportional
+	event?: EarningRuleEvent
+	custom_event?: {
+		schema_id: string
+	}
+	segment?: {
+		id: string
+	}
+	source: {
+		banner?: string
+		object_id: string
+		object_type: string
+	}
+	loyalty_tier?: {
+		id: string
+	}
+	object: 'earning_rule'
+	automation_id: string
+	start_date?: string
+	expiration_date?: string
+	validity_timeframe?: {
+		duration: string
+		interval: string
+	}
+	validity_day_of_week?: number[]
+	metadata: Record<string, any>
+}
+
+export type EarningRuleEvent =
+	| 'order.paid'
+	| 'customer.segment.entered'
+	| 'custom_event'
+	| 'customer.loyalty.tier.upgraded'
+	| 'customer.loyalty.tier.downgraded'
+	| 'customer.loyalty.tier.prolonged'
+	| 'customer.loyalty.tier.joined'
+	| 'customer.loyalty.tier.left'
+
+export interface EarningRuleLoyaltyFixed {
```

Comment:
I just wonder about the naming - do we need the word `Loyalty` in `EarningRuleLoyaltyFixed` interface name? The same with all names below...